### DISCUSSION TITLE
Create or update many

### DISCUSSION THREAD

**Discussion by JordanGibson** [prisma/prisma#16618] [Q&A] [answered]
Hi all,

I have a scenario where I am making external API calls, passing the response and storing it in my database with prisma. The IDs (PKs) which I am using match that on the external API. The IDs are guaranteed to remain the same, but the data with the objects are not. I am looking for the most eloquent way to insert all records which do not already exist (by PK ID) and if a record already exists, update its data.

I don’t believe upsert accomplishes this behaviour as a query would need performed for every record, which seems inefficient. What would be the cleanest way to perform an “upsertMany” by primary key?

I’m using Postgres, but I’m not overly sure if that matters. Thanks!

**Comment by nurul3101** [comment, marked as answer]
Hello @JordanGibson 👋 

We have a Feature request of `upsertMany` here: [#10362](https://github.com/prisma/prisma/issues/10362)
If you can leave a 👍 to the feature request, then it would help our product team in prioritising it.
There are some helpful comments in the Feature Request. You can use Raw Query as an [alternative](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) and achieve this through `INSERT ... ON DUPLICATE KEY UPDATE` sql statement.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
